### PR TITLE
update to latest Firebase API v3

### PR DESCRIPTION
**Pull request by continuata**
Firebase has changed its entire API as part of the move to integrate more fully with Google infrastructure.
There are a number of breaking changes, especially with regard to how to login works using the service Account.
Here are changes that make flashlight work for me in this PR ... probably also need some documentation to show how to get these extra environment params.
(some of them are a bit tricky and require adding users to IAM roles in the new Firebase google project)

### CODE REVIEW

**Comment by continuata** [comment]
I signed it!

**Comment by katowulf** [comment]
Thanks Stu! Looks good after a quick scan. I should be able to test and merge this soon.

**Comment by continuata** [comment]
Has there been any movement on this yet?

File: config.js [hunk 1]
```diff
@@ -10,18 +10,30 @@
...
-exports.FB_TOKEN = process.env.FB_TOKEN || null;
+exports.FB_URL   = 'https://' + process.env.FB_NAME + '.firebaseio.com';
 
 // The path in your Firebase where clients will write search requests
 exports.FB_REQ   = process.env.FB_REQ || 'search/request';
 
 // The path in your Firebase where this app will write the results
 exports.FB_RES   = process.env.FB_RES || 'search/response';
 
+// See https://firebase.google.com/docs/web/setup#project_setup for how to
```
Comment (katowulf) [inline config.js]: The appropriate doc link here would be https://firebase.google.com/docs/server/setup.

File: lib/DynamicPathMonitor.js [hunk 2]
```diff
@@ -9,16 +9,16 @@ function DynamicPathMonitor(ref, factory) {
 
 DynamicPathMonitor.prototype = {
   _add: function(snap) {
-    var name = snap.key();
-    var pathProps = snap.val();
+    var name = snap.key;
+    var pathProps = snap.val;
```
Comment (katowulf) [inline lib/DynamicPathMonitor.js]: `snap.key` should change here, but `snap.val()` is still correct.

File: config.js [hunk 3]
```diff
@@ -10,18 +10,30 @@
...
+// auto-generate this config
+exports.FB_SERVICEACCOUNT = {
```
Comment (katowulf) [inline config.js]: So I'm unclear why this is needed. I'd prefer to see the convention established [here](https://firebase.google.com/docs/server/setup#initialize_the_sdk) and simply include the serviceAccountCredentials as a path to the downloaded file.

**Comment by continuata** [comment]
let me know if that's better, seems to work on my heroku instance

**Comment by diego1686** [comment]
Any news?

**Comment by sweetcoco** [comment]
@katowulf  This is a solution but it requires you to add your fb credentials to git, something that isn't exactly ideal. Would it be better to submit the credentials object as a config var? Or am I missing something?

**Comment by katowulf** [comment]
This is a closed PR. Submit an issue if you want to discuss.

**Pull request merged**